service/src/main/java/uk/gov/hmcts/reform/fpl/testingsupport/controllers/TestingSupportController.java

```diff
@@ -168,6 +169,32 @@ public NotificationList getEmails(@PathVariable("caseId") String caseId) {
         }
     }
 
+    @PostMapping("/testing-support/case/{caseId}/revoke-access")
+    public void revokeCaseAccess(@PathVariable("caseId") Long caseId, @RequestBody Map<String, String> requestBody) {
+        String userToken = idamClient.getAccessToken(userConfig.getUserName(), userConfig.getPassword());
+
+        final String email = requestBody.get("email");
+        final String password = requestBody.get("password");
+        final String role = requestBody.get("role");
+
+        log.info("About to revoke {} from user {} to case {}", role, email, caseId);
+
+        final String token = idamClient.getAccessToken(email, password);
+        final String userId = idamClient.getUserDetails(token).getId();
+
+        final CaseAssignedUserRolesRequest accessRequest = CaseAssignedUserRolesRequest.builder()
+            .caseAssignedUserRoles(List.of(CaseAssignedUserRoleWithOrganisation.builder()
+                .caseDataId(caseId.toString())
+                .userId(userId)
+                .caseRole(role)
+                .build()))
+            .build();
+
+        caseAccess.removeCaseUserRoles(userToken, authToken.generate(), accessRequest);
+
+        log.info("Role {} revoked from user {} to case {}", role, email, caseId);
```

Comment:
Why do we need this for document removal? Should we separate it to another PR / ticket?